cmd/agent/common/helpers.go

```diff
@@ -49,3 +51,34 @@ func setupConfig(confFilePath string, configName string, withoutSecrets bool) er
 	}
 	return nil
 }
+
+// SetupSystemProbeConfig adds the system-probe.yaml file to the config object
+func SetupSystemProbeConfig(sysProbeConfFilePath string) error {
+	// set the paths where a config file is expected. Without this, if there is a system-probe.yaml
+	// in the default location it will overwrite whatever custom path is passed in
+	if len(sysProbeConfFilePath) != 0 {
+		// If they pass in the file directly, assume we should use it
+		if strings.HasSuffix(sysProbeConfFilePath, ".yaml") {
+			config.Datadog.SetConfigFile(sysProbeConfFilePath)
+		} else {
+			config.Datadog.SetConfigFile(sysProbeConfFilePath + "system-probe.yaml")
+		}
+	} else {
+		// Assume it is in the default location if nothing is passed in
+		config.Datadog.SetConfigName("system-probe")
+		config.Datadog.SetConfigType("yaml")
+		config.Datadog.AddConfigPath(DefaultConfPath)
+	}
+
+	if err := config.Datadog.ReadInConfig(); err != nil {
+		return err
+	}
+
+	// The full path to the location of the unix socket where connections will be accessed
+	// This is not necessarily set in the system-probe.yaml, so set it manually if it is not
+	if !config.Datadog.IsSet("system_probe_config.sysprobe_socket") {
+		config.Datadog.Set("system_probe_config.sysprobe_socket", proc_config.GetSocketPath())
```

Comment:
For the rest of the agent env var have the priority over files. we should do the same here I think.

Reply:
I've edited the code here to ensure that env vars are loaded, so this should be dealt with